control-plane/plugin/src/resources/node.rs

```diff
@@ -118,7 +118,14 @@ impl ListWithArgs for Nodes {
             Ok(nodes) => {
                 let node_display =
                     NodeDisplayLabels::new_nodes(nodes.clone().into_body(), args.show_labels());
-                print_table(output, node_display);
+                match output {
+                    OutputFormat::Yaml | OutputFormat::Json => {
+                        print_table(output, node_display.inner);
+                    }
+                    OutputFormat::None => {
+                        print_table(output, node_display);
```

Comment:
Doesn't the print table do the print based on formats? Also why are we passing inner separately?

Reply:
Could we not have passed the inner to the print_table only?

Reply:
In case of json and yaml the serde can't  deserialize the struct. 
The similar changes are done in our code base earlier 
Refer [here](https://github.com/openebs/mayastor-control-plane/blob/1ee5592f1df0ce1d5e36c360315fa4c327361527/control-plane/plugin/src/resources/node.rs#L484)

Reply:
Alright.

Reply:
I think the problem is actually in the NodeDisplay struct which should not have flatten on the vector directly.

Reply:
I tried the same way as its done in  [`pub struct NodeDisplay {`](https://github.com/openebs/mayastor-control-plane/blob/919e1a1afbfcbc76ebe08035889d59d2ff12ae61/control-plane/plugin/src/resources/node.rs#L380)

The node display format uses `node_display_print` which does the same as this PR